carto_auth/auth.py

```diff
@@ -108,7 +106,7 @@ def from_oauth(
         data = get_oauth_token_info(open_browser)
         return cls(
             mode=mode,
-            api_base_url=get_api_base_url(data.get("access_token")),
+            api_base_url=api_base_url or get_api_base_url(data.get("access_token")),
```

Comment:
Just for curiosity why when using cache we don't do the same override?

Reply:
The api_base_url is cached, so mixing the cached keys with a different api_base_url can generate some errors